### PR TITLE
NAS-127003 / 24.10 / Refactoring Replication List

### PR DESCRIPTION
**Pull request by denysbutenko**
Changes:

- Added paginator for cloudsync task list for consistency.
- Removed `ReplicationTaskUi` and updated `ReplicationTaskCard` component
- Refactored `ReplicationListComponent` to use ix-table2

Testing:

- Ensure no regressions

### CODE REVIEW

File: src/app/pages/data-protection/replication/replication-list/replication-list.component.ts [hunk 1]
```diff
@@ -1,306 +1,280 @@
...
+    this.filterString = query.toLowerCase();
+    this.dataProvider.setRows(this.replicationTasks.filter((task) => {
+      return task.name.includes(this.filterString);
+    }));
+  }
+
+  columnsChange(columns: typeof this.columns): void {
+    this.columns = [...columns];
+    this.cdr.detectChanges();
+    this.cdr.markForCheck();
```
Comment (RehanY147) [inline src/app/pages/data-protection/replication/replication-list/replication-list.component.ts:241]: Only one of these is needed.

File: src/app/pages/data-protection/replication/replication-list/replication-list.component.ts [hunk 2]
```diff
@@ -1,306 +1,280 @@
...
+  }
+
+  downloadKeys(row: ReplicationTask): void {
+    const fileName = `${row.name}_encryption_keys.json`;
+    this.ws.call('core.download', ['pool.dataset.export_keys_for_replication', [row.id], fileName])
+      .pipe(this.appLoader.withLoader(), untilDestroyed(this))
+      .subscribe({
+        next: ([, url]) => {
+          const mimetype = 'application/json';
+          this.storage.streamDownloadFile(url, fileName, mimetype).pipe(untilDestroyed(this)).subscribe({
```
Comment (RehanY147) [inline src/app/pages/data-protection/replication/replication-list/replication-list.component.ts:251]: Prefer using `switchMaps` pipes to keep the flow more readable. Something like the following
`this.ws.call(...).pipe( ... , switchMap(([, url]) => { ... ; return this.storage.streamDownloadFile(...); ... }))`

**Review by RehanY147 — APPROVED** [review]
Works. Just consider the suggested code improvements.

**Comment by denysbutenko** [comment]
@RehanY147 Thanks for the feedback. Suggestions make sense, but they would break consistency with other components (e.g., see `ReplicationTaskCardComponent`) - I'd prefer to keep it as is for now. We could continue the discussion to define common patterns for the `ix-table2`. For example, remove the wrapper on `dataProvider.load()` in every component or make table actions reusable between components).

**Comment by denysbutenko** [comment]
I created a ticket to keep track - https://ixsystems.atlassian.net/browse/NAS-127168

**Pull request merged**